### PR TITLE
Add join tables

### PR DESCRIPTION
**Pull request by NitBravoA92**
# 🎏 For this Milestone, I

✅ Create the `vets` table with four columns: `id`, `name`, `age`, and `date_of_graduation`

✅ Create the `specializations` table with two columns: `vet_id`, and `species_id`

✅ Create the `visits` table with three columns: `animal_id`,  `vet_id` and `date_of_visit`

✅ Insert data into the vets' table

✅ Insert data into the specializations' table

✅ Insert data into the visits' table

✅ Create the necessary queries (using the JOINs clauses and Sub-queries) to answer each of the questions listed in the requirements.

The following screenshots show the result of all the queries made to the join tables, according to the project requirements

### SCREENSHOTS: 

- Who was the last animal seen by William Tatcher?
query_01

- How many different animals did Stephanie Mendez see?
query_02

- List all vets and their specialties, including vets with no specialties.
query_03

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
query_04

- What animal has the most visits to vets?
query_05

- Who was Maisy Smith's first visit?
query_06

- Details for most recent visit: animal information, vet information, and date of visit.
query_07

- How many visits were with a vet that did not specialize in that animal's species?
query_08

- What specialty should Maisy Smith consider getting? Look for the species she gets the most?
query_09

# ⭐To the code reviewer 👨‍💻

💝 I really thank 🙇‍♂️ you, dear code reviewer 👨‍🎨 for dedicating your precious 🥇 time ⌚ to check the PR 🧾 .

I kindly ask 🙏 dear code reviewer 🤵 that if there is any kind of issue 🦯 in this project, please do list 📃 them in a descriptive 💡 manner and give your best suggestions 🎁 if needed.

If you think some big issues are essential to be changed ♻️ please kindly contact 📞 me through Zoom or Slack, I am available on Microverse UTC-6 time zone.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -199,3 +199,44 @@ SELECT o.full_name AS animal_owner, a.name AS animal_name, a.escape_attempts AS
...
+-- Question 7: Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.*, ve.id AS vet_id, ve.name AS vet_name, ve.age AS vet_age, ve.date_of_graduation AS vet_graduation_date, v.date_of_visit AS visit_date FROM visits v INNER JOIN animals a ON v.animal_id = a.id INNER JOIN vets ve ON v.vet_id = ve.id ORDER BY visit_date DESC;
+
+
+-- Question 8: How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS total_visits_no_specialization FROM visits v INNER JOIN animals a ON v.animal_id = a.id WHERE v.vet_id NOT IN (SELECT s.vet_id FROM specializations s WHERE s.species_id = a.species_id);
+
+
+-- Question 9: What specialty should Maisy Smith consider getting? Look for the species she gets the most?
+SELECT (SELECT s.name FROM species s WHERE s.id = a.species_id) AS animal_specie, COUNT(*) AS total_visits FROM visits v INNER JOIN vets ve ON v.vet_id = ve.id INNER JOIN animals a ON v.animal_id = a.id WHERE ve.name = 'Maisy Smith' GROUP BY animal_specie ORDER BY total_visits DESC;
```
Comment (abdulhamiid) [inline queries.sql]: image

Please add the screenshots of the correct result for the query above in your pull request description. 🛠🤗

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @NitBravoA92 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Understand the different types of relationships between tables.
- ✔️ Prepare complex queries that answer analytical questions.
- ✔️ Use primary key & foreign key mechanisms for joining tables.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (NitBravoA92) [reply queries.sql]: @abdulhamiid  Hello. Thank you for taking the time to review my project. Please, check the final part of the screenshot in the pull request, it shows the result of the query you are asking for.

Captura desde 2023-08-17 23-08-13

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @NitBravoA92 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Join table is added for visits
-  ✅ Great work using the right GitFlow 
-  ✅ Screenshots are added
-  ✅ Successful queries

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**